typescript/src/index.ts

```diff
@@ -0,0 +1,336 @@
+        await query.delete()
+        logInfo(`Query ${queryName} deleted.\n`)
+    } catch (err) {
+        logInfo(`Query ${queryName} does not exist.\n`)
+    }
+    return account.addQuery(queryName, params)
+}
+
+async function addJobQueries(account: Account, source: Dataset) {
+    const files = (await readdir(QUERY_PATH))
+        // Only use sparql files
+        .filter(f => extname(f) === '.sparql')
+
+    const queries = []
+    for (const file of files) {
+        const filePath = join(QUERY_PATH, file)
+        const queryString = await readFile(filePath, 'utf8')
+        const queryName = parse(filePath).name
+
+        const params: AddQueryOptions = {
+            dataset: source,
+            queryString,
+            serviceType: 'speedy',
+            output: 'response',
+            variables: [
+                {
+                    name: 'since',
+                    required: false,
+                    termType: 'Literal',
+                    datatype: 'http://www.w3.org/2001/XMLSchema#dateTime',
+                }
+            ]
+        }
+
+        const query = await addQuery(account, queryName, params)
+        queries.push(query)
+    }
+    return queries
+}
+
+async function createTableNode(tableName: string): Promise<TableNode> {
+    const tableInfo = new TableInfo(tableName)
+    const tableNode = {
+        tableInfo,
+        tempTable: await createTempTable(tableInfo),
+        // Get the actual columns from the database
+        columns: await getTableColumns(tableInfo),
+        // Get dependent tables from the database
+        dependencies: await getDependentTables(tableInfo),
+        // Get primary keys
+        primaryKeys: await getTablePrimaryKeys(tableInfo)
+    }
+    tableIndex.addNode(tableName, tableNode)
+    return tableNode
+}
+
+
+async function processBatch(tableName: string, batch: Array<Record<string, string>>) {
+    // Create temp table if not exists
+    const tableNode = tableIndex.hasNode(tableName) ? tableIndex.getNodeData(tableName) : await createTableNode(tableName)
+    await batchInsertUsingCopy(tableNode, batch)
+}
+
+// Process each record and add it to the appropriate batch
+async function processRecord(
+    subject: string,
+    record: Record<string, string>,
+    tableName: string,
+    batches: { [tableName: string]: Array<Record<string, string>> }
+) {
+    logDebug(`Process record for ${subject}: ${JSON.stringify(record)}`)
+
+    if (!record || !tableName || !subject) return
+
+    if (!batches[tableName]) {
+        batches[tableName] = []
+    }
+
+    batches[tableName].push(record)
+
+    if (batches[tableName].length >= BATCH_SIZE) {
+        logDebug(`Maximum batch size reached for ${tableName}; processing.`)
+        const batch = batches[tableName]
+        await processBatch(tableName, batch)
+        batches[tableName] = []
+    }
+}
+
+// Main function to parse and process the gzipped TriG file from a URL
+async function processGraph(graph: Graph) {
```

Comment:
This part looks like it works, but it is quite complicated, so it might need some more comments. e.g. it is not really clear in this function `batches` is updated somewhere. Maybe it could be more clear if processRecord returns `batches` and processBatch is called in processGraph?